### DISCUSSION TITLE
Using "grid" without specifying the height of "swiper"

### DISCUSSION THREAD

**Discussion by vasilueff** [nolimits4web/swiper#7366] [Q&A]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

Cant fork it

### Bug description

There is a problem when using "grid" without specifying the height of "swiper". The height of each ".swiper-slide" is calculated as a percentage of the total height of the ".swiper-wrapper". And the script does not count the total height of the ".swiper-wrapper".
Because of this, all items are shown in one column.

![1710217502440](https://github.com/nolimits4web/swiper/assets/7285629/bd664806-932a-4db6-8d2c-74faa037f619)

```
  const swiperReasons = new Swiper('.swiper-reasons', {
    loop: false,
    grabcursor: true,
    spaceBetween: 0,
    breakpoints: {
        260: {
            slidesPerView: 1,
            loop: true,
            grid: {
                rows: 2,
                },

        },
        768: {
            slidesPerView: 2,
            grid: {
                rows: 4,
                },

        },
    },

  // Navigation arrows
  navigation: {
    nextEl: '.swiper-reasons-btn-next',
    prevEl: '.swiper-reasons-btn-prev',
  },
});
```

### Swiper version

11

### Platform/Target and Browser Versions

All browsers

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a Swiper issue and not a framework-specific issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

**Comment by alexej-d** [comment]
Same here. Using { grid: { fill: 'row' } } works better, but is sometimes not preferable.. This also defeats the purpose of the rows count, when in column mode it is simply ignored

**Comment by joelthorner** [comment]
I have the same problem. I want a grid of 4 slides in vertical and in row, it doesn't order them well and in column I can't know the total height as they are slides with variable height. 😢

**Comment by wowblvck** [comment]
I have a same problem too. Is there any solution? @nolimits4web

**Comment by FrankJingZhi** [comment]
So is there a solution to this problem?

**Reply by joelthorner** [reply]
<img src="https://media1.giphy.com/media/qBvMJ8SxTRQiohH9kj/giphy.gif"/>

**Comment by Pfenya** [comment]
I found this older comment in an old issue and that solved it for my use case at least.

```
.swiper-wrapper {
      display: grid;
      grid-template-columns: 1fr;
      grid-template-rows: auto auto; // here is the amount of slidesPerColumn you want
      grid-auto-flow: column;

      & > .swiper-slide {
        height: fit-content;
      }
    }
```
    

https://github.com/nolimits4web/swiper/issues/3754#issuecomment-774240879

**Reply by rtcpw** [reply]
Sadly, this doesn't work well when using `spaceBetween` swiper prop. I suppose it can be worked around as well, but the question is where will the workarounds end? @nolimits4web
